lib/xboxkrnl/xboxkrnl.h

```diff
@@ -895,12 +895,12 @@ typedef struct _KQUEUE
     LIST_ENTRY ThreadListHead;
 } KQUEUE, *PKQUEUE, *RESTRICTED_POINTER PRKQUEUE;
 
-struct KTHREAD;
+struct _KTHREAD;
```

Comment:
Why is this rename required?

Reply:
Struct above and below follow the same naming scheme. :+1:

Reply:
In C++ `struct KTHREAD` declares a type called `KTHREAD`; whereas in C you'd still have to use `struct KTHREAD` to refer to the same thing (not sure about terminology here, but I believe you'd "prototype" the struct instead of "declaring" it?).

Hence you'd typically prefix the tag in C and use a typedef like `typedef struct _KTHREAD { ... } KTHREAD;`, so both, C and C++ can just use `KTHREAD` when referring to the same type (instead of C++ using `KTHREAD` but C using `struct KTHREAD`).

Reply:
Ah yes. Struct identifiers should have underscore prefix. Type definitions via typedef should not. I was asking if there was a functional reason for this change, not just convention. IIRC C++ should allow the names to be the same.

> whereas in C you'd still have to use struct KTHREAD to refer to the same thing
You should be able to use the typedef in both C in C++ ofc.